hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/ServerRpcConnection.java

```diff
@@ -716,14 +717,22 @@ public String get(RPCTInfo carrier, String key) {
       ServerCall<?> call = createCall(id, this.service, md, header, param, cellScanner,
         totalRequestSize, this.addr, timeout, this.callCleanup);
 
-      if (!this.rpcServer.scheduler.dispatch(new CallRunner(this.rpcServer, call))) {
+      if (this.rpcServer.scheduler.dispatch(new CallRunner(this.rpcServer, call, span))) {
+        // unset span do that it's not closed in the finally block
+        span = null;
```

Comment:
Yes, the code structure here is not friendly for tracing...

Thanks for the detailed comments!